include/extractor/guidance/intersection_normalizer.hpp

```diff
@@ -57,12 +60,25 @@ class IntersectionNormalizer
 
     // check if two indices in an intersection can be seen as a single road in the perceived
     // intersection representation. See below for an example. Utility function for
-    // MergeSegregatedRoads
+    // MergeSegregatedRoads. It also checks for neighboring merges.
+    // This is due possible segments where multiple roads could end up being merged into one.
+    // We only support merging two roads, not three or more, though.
+    //       c                                  c
+    //      /                                  /
+    // a - b     -> a - b - (c,d) but not a - b d   -> a,b,(cde)
+    //      \                                  \
+    //       d                                  e
     bool CanMerge(const NodeID intersection_node,
                   const Intersection &intersection,
                   std::size_t first_index,
                   std::size_t second_index) const;
 
+    // A tool called by CanMerge
+    bool InnerCanMerge(const NodeID intersection_node,
+                       const Intersection &intersection,
+                       std::size_t first_index,
+                       std::size_t second_index) const;
```

Comment:
How about the explanation at CanMerge? I've added a few additional remarks here, but what was missing from the above description for you that you couldn't tell the difference? Since it's internal, there is no danger of being called from someone